v2.1/online-schema-changes.md

```diff
@@ -0,0 +1,205 @@
+
+Benefits of online schema changes include:
+
+- Changes to your table schema happen while the database is running.
+- The schema change runs as a [background job][show-jobs] without holding locks on the underlying table data.
+- Your application's queries can run normally, with no effect on read/write latency. The schema is cached for performance.
+- Your data is kept in a safe, [consistent][consistent] state throughout the entire schema change process.
+
+{{site.data.alerts.callout_success}}
+Support for schema changes within transactions is limited. We recommend doing schema changes outside transactions where possible.  When a schema management tool uses transactions on your behalf, we recommend only doing one schema change operation per transaction.
+{{site.data.alerts.end}}
+
+{{site.data.alerts.callout_info}}
+Schema changes keep your data consistent at all times, but they do not run inside user transactions in the general case. This is necessary so the cluster can remain online and continue to service application reads and writes. For more information, see [Limitations](#limitations).
+{{site.data.alerts.end}}
+
+<span class="version-tag">New in v2.1:</span> You can run schema changes inside the same transaction as a [`CREATE TABLE`][create-table] statement. For more information, [see this example](#run-schema-changes-inside-a-transaction-with-create-table).
+
+## How online schema changes work
+
+At a high level, online schema changes are accomplished by using a bridging strategy involving concurrent uses of multiple versions of the schema. Schema changes are run as a series of transactions:
+
+1. A user-level transaction initiated by [`ALTER TABLE`][alter-table], [`CREATE INDEX`][create-index], [`TRUNCATE`][truncate], etc. The user-level transaction schedules the schema change for the schema change engine to complete.
+
+2. A series of system-level transactions calculated by the schema change engine that convert the original schema to the new schema in discrete steps while ensuring that the underlying table data is always in a consistent state. These transactions are executed as a [background job][show-jobs].
+
+This approach allows the schema change engine to roll out a new schema while the previous version is still in use. It then backfills or deletes the underlying table data as needed in the background, while the cluster is still running and servicing reads and writes from your application.
+
+During the backfilling process, the schema change engine updates the underlying table data to make sure all instances of the table are stored according to the requirements of the new schema.
+
+Once backfilling is complete, all nodes will switch over to the new schema, and will allow reads and writes of the table using the new schema.
+
+For more technical details, see [How online schema changes are possible in CockroachDB][blog].
+
+## Limitations
```

Comment:
This is great, but I think there may be a few other limitations to mention here, based on http://www.cockroachlabs.com/docs/dev/known-limitations.html#new-limitations. Ideally, we would have one include that is used both here and on the known limitations page.

Reply:
Good point!  Fixed by creating includes of the schema change portions of the Known Limitations page, and including them here.

That led to rewriting this section to be able to house everything in a hopefully coherent way, in the following subsections:

- Overview
- No schema changes within transactions
- No schema changes between executions of prepared statements
- Examples of statements that fail

See update for details.